sphinx/builders/epub3.py

```diff
@@ -157,30 +169,23 @@ def build_navlist(self, nodes):
                 continue
             if node['level'] > self.config.epub_tocdepth:
                 continue
-            if node['level'] == level:
-                navlist.append(self.new_navlist(node, level))
-            elif node['level'] == level + 1:
-                navstack.append(navlist)
-                navlist = []
-                level += 1
-                if lastnode and self.config.epub_tocdup:
```

Comment:
It seems your new code does not refer `epub_tocdup`.
Could you add testcases to confirm this PR does not change the behavior?